final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,294 @@
+import requests
+import argparse
+import json
+import logging
+import socket
+import bs4
+from bs4 import BeautifulSoup
+import CustomException
+from datetime import datetime
+import converter
+import colorizer
+LOG = logging.getLogger("LOG")
+
+
+def is_connected() -> bool:
+    LOG.info("Checking internet connection...")
+    try:
+        socket.create_connection(("www.google.com", 80))
+    except Exception:
+        raise CustomException.ConnectionError
+    return True
+
+
+def find_news(url) -> bs4.element.ResultSet:
+    """Parses news"""
+    logging.info("Searching for news on " + url)
+    try:
+        resp = requests.get(url)
+        soup = BeautifulSoup(resp.content, features="xml")
+        items = soup.findAll('item')
+        LOG.info(f'{len(items)} pieces of news have been found')
+    except requests.exceptions.MissingSchema:
+        raise CustomException.WrongUrl
+    except requests.exceptions.ConnectionError:
+        LOG.error("Can't connect to the " + url)
+        raise CustomException.UrlUnreachable
+    return items
+
+
+def find_channel(url) -> bs4.element.Tag:
+    logging.info("Getting feed from " + url)
+    try:
+        resp = requests.get(url)
+        soup = BeautifulSoup(resp.content, features="xml")
+        channel = soup.find('channel')
+    except requests.exceptions.ConnectionError:
+        raise CustomException.WrongUrl
+    except requests.exceptions.InvalidURL:
+        raise CustomException.WrongUrl
+    except requests.exceptions.MissingSchema:
+        raise CustomException.WrongUrl
+    return channel
+
+
+def collect_news(items, limit) -> list:
```

Comment:
Same as above, this function is large which lowers code readability. It is better to somehow split it.